### PR TITLE
Setting up testing infrastructure

### PR DESCRIPTION
**Pull request by Marketionist**
**This pull request adds front-end tests infrastructure**

Details:
- [x] adds ```.editorconfig```, ```.jscsrc```, ```.jshintrc```
- [x] updates ```.gitignore```
- [x] adds ```browserslist```
- [x] adds ```gulpfile.js```
- [x] adds ```package.json```
- [x] adds ```test_requirements/runserver.sh```
- [x] updates ```tox.ini```
- [x] updates ```.travis.yml```
- [x] adds basic configuration for unit tests
- [x] adds basic configuration for integration tests

### CODE REVIEW

File: aldryn_faq/tests/frontend/fixtures/faq.html [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<div class="faq">
```
Comment (FinalAngel) [inline aldryn_faq/tests/frontend/fixtures/faq.html]: write on one line

Comment (Marketionist) [inline aldryn_faq/tests/frontend/fixtures/faq.html]: Done

File: .travis.yml [hunk 2]
```diff
@@ -24,7 +26,6 @@ matrix:
       env: DJANGO=1.7 CMD=./test
   fast_finish: true
   include:
-    # Change this to Django 1.7 once it is working.
     - python: 2.7
       env: DJANGO=1.7 CMD="flake8 aldryn_newsblog"
```
Comment (FinalAngel) [inline .travis.yml]: newsblog?

**Comment by FinalAngel** [comment]
one last comment otherwise LGTM

File: .travis.yml (hunk 2 above)
Comment (Marketionist) [inline .travis.yml]: Yes, looked strange for me too. Maybe @mkoistinen knows something about this?

**Comment by vxsx** [comment]
js linting fails

**Pull request merged**